wallet/src/repository/wallet/mod.rs

```diff
@@ -918,28 +995,20 @@ where
         }: types::VttParams,
     ) -> Result<types::VTTransaction> {
         let mut state = self.state.write()?;
-        let components =
-            self.create_vt_transaction_components(&mut state, value, fee, Some((pkh, time_lock)))?;
+        let (inputs, outputs) = self.create_vt_transaction_components(
+            &mut state,
+            vec![ValueTransferOutput {
+                pkh,
+                value,
+                time_lock,
+            }],
+            fee,
+        )?;
 
-        let body = types::VTTransactionBody::new(components.inputs, components.outputs);
+        let body = types::VTTransactionBody::new(inputs.clone(), outputs);
         let sign_data = body.hash();
-        let signatures: Result<Vec<types::KeyedSignature>> = components
-            .sign_keys
-            .into_iter()
-            .map(|sign_key| {
-                let public_key = From::from(types::PK::from_secret_key(&self.engine, &sign_key));
-                let signature = From::from(types::signature::sign(
-                    &self.engine,
-                    sign_key,
-                    sign_data.as_ref(),
-                )?);
-
-                Ok(types::KeyedSignature {
-                    signature,
-                    public_key,
-                })
-            })
-            .collect();
+        let signatures =
+            self.create_signatures_from_inputs_and_update_balance(inputs, sign_data, &mut state);
```

Comment:
This says "update_balance", but I think the old version of `create_vtt` did not update the balance?

Reply:
Ah, previously the unconfirmed balance was being updated in `create_transaction_components`. Now `create_transaction_components` does not update the balance.

Reply:
It modifies an mutable balance, so I think it updates

Reply:
> Did you test it?

No. But there should be tests for that already, somewhere?

Reply:
removed it, because it really didnt update the balance